source/_integrations/template.markdown

```diff
@@ -322,17 +321,19 @@ sensor:
 
 The `template` sensors are not limited to use attributes from other entities but can also work with [Home Assistant's template extensions](/docs/configuration/templating/#home-assistant-template-extensions).
 
-This template contains no entities that will trigger an update (as `now()` is a function), so we add an `entity_id:` line with an entity that will force an update - here we're using a [date sensor](/integrations/time_date) to get a daily update:
+This template contains no entities that will trigger an update, so we add in a reference to the `sensor.date` sensor from the [Time & Date](/integrations/time_date/), which will update once a day.
 
 {% raw %}
 
 ```yaml
 sensor:
+  - platform: time_date
+    display_options:
+      - 'date'
   - platform: template
     sensors:
       nonsmoker:
-        value_template: "{{ (( as_timestamp(now()) - as_timestamp(strptime('06.07.2018', '%d.%m.%Y')) ) / 86400 ) | round(2) }}"
-        entity_id: sensor.date
+        value_template: '{{ states('sensor.date').state and (( as_timestamp(now()) - as_timestamp(strptime("06.07.2018", "%d.%m.%Y")) ) / 86400 ) | round(2) }}'
         friendly_name: 'Not smoking'
         unit_of_measurement: "Days"
```

Comment:
This template doesn't work.   `states('sensor.date')` returns a string and not an object so `.state` doesn't work.   Pasting this into the template editor results in a blank.

I take it that `{{ states('sensor.date').state and` is just there to get the parser to extract the entity_id?

Reply:
It should be `states('sensor.date')`... fixed